geode/src/test/scala/akka/stream/alpakka/geode/scaladsl/GeodeContinuousSourceSpec.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.geode.scaladsl
+
+import akka.NotUsed
+import akka.stream.scaladsl.{Flow, Sink}
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.Await
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
+
+class GeodeContinuousSourceSpec extends GeodeBaseSpec {
+
+  private val log = LoggerFactory.getLogger(classOf[GeodeContinuousSourceSpec])
+
+  "Geode continuousQuery" should {
+    it { geodeSettings =>
+      "retrieves continuously elements from geode" in {
+
+        //#connection-with-pool
+        val reactiveGeode = new ReactiveGeode(geodeSettings) with PoolSubscription
+        //#connection-with-pool
+
+        val flow: Flow[Person, Person, NotUsed] = reactiveGeode.flow(personsRegionSettings)
+
+        //#continuousQuery
+        val source =
+          reactiveGeode
+            .continuousQuery[Person]('test, s"select * from /persons")
+            .runWith(Sink.fold(0) { (c, p) =>
+              log.debug(s"$p $c")
+              if (c == 19) {
+                reactiveGeode.closeContinuousQuery('test).foreach { _ =>
+                  log.info("test cQuery is closed")
+                }
```

Comment:
Is there a reason such style has been chosen? A more stream-like approach to take a finite number of stream elements from a never ending continuous stream would be:

    reactiveGeode.continuosQuery(...).take(19).runWith(Sink.fold(...))